### PR TITLE
Feature/cgo invoker

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Create CgoInvoker with Invoker Interface.

### Benefits
Unified interface, can be replaced with builtin actors.

### Possible Drawbacks 
no

### CODE REVIEW

File: core/vm/runtime/impl/env.cpp [hunk 1]
```diff
@@ -200,6 +201,7 @@ namespace fc::vm::runtime {
     execution->gas_limit = message.gas_limit;
     execution->origin = message.from;
     execution->origin_nonce = message.nonce;
+    execution->invoker = std::move(invoker);
```
Comment (turuslan) [inline core/vm/runtime/impl/env.cpp:204]: Don't `move` here

File: core/vm/actor/cgo/cgo_invoker.hpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#include "vm/actor/invoker.hpp"
+
+namespace fc::vm::actor::cgo {
+
+  /**
+   * Invoker implementation that invokes GoLang specs-actors using cgo
+   */
+  class CgoInvoker : public Invoker {
+   public:
+    explicit CgoInvoker(bool test_vectors);
```
Comment (turuslan) [inline core/vm/actor/cgo/cgo_invoker.hpp:18]: Should `InvokerImpl` have same option?

File: core/vm/runtime/impl/env.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply core/vm/runtime/impl/env.cpp:204]: ok

File: core/vm/actor/cgo/cgo_invoker.hpp (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply core/vm/actor/cgo/cgo_invoker.hpp:18]: test_vectors is used only for randomness, and randomness is provided by `Runtime`. I think it should be removed from `CgoInvoker` to `Runtime`, and `InvokerImpl` shouldn't have the option.

**Review by turuslan — APPROVED** [review]

File: core/vm/actor/cgo/cgo_invoker.hpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    void config(const StoragePower &min_verified_deal_size,
+                const StoragePower &consensus_miner_min_power,
+                const std::vector<RegisteredProof> &supported_proofs) override;
+
+    outcome::result<InvocationOutput> invoke(const Actor &actor,
+                                             Runtime &runtime) override;
+
+   private:
+    bool _test_vectors{false};
```
Comment (wer1st) [inline core/vm/actor/cgo/cgo_invoker.hpp:28]: For what? You already defined it in the constructor's initialisation list.
Moreover it is not a good practice to define members value in .h files

Comment (Alexey-N-Chernyshov) [reply core/vm/actor/cgo/cgo_invoker.hpp:28]: fixed

**Comment by Alexey-N-Chernyshov** [comment]
reopen PR as trunk/cpp-actors

**Pull request closed without merging**